resources/js/Components/BaseInput.vue

```diff
@@ -0,0 +1,44 @@
+<script setup lang="ts">
+
+type InputType = 'text' | 'email' | 'password' | 'number' | 'search';
+
+defineProps<{
+  modelValue: string
+  label?: string
+  type?: InputType
+  name: string
+  id: string
+  placeholder?: string
+  error?: string | null
+}>()
+
+defineEmits<{
+  'update:modelValue': [value: string]
+}>()
+</script>
+
+<template>
+  <label class="block text-base text-gray-500 mb-1" :for="id">
+    {{ label }}
+  </label>
+  <div class="relative">
+    <input
+      :id="id"
+      :name="name"
+      :type="type"
+      :placeholder="placeholder"
+      :value="modelValue"
+      :class="[
+        'w-full h-12 px-4 pr-10 font-medium rounded-lg focus:outline-none text-Brand hover:bg-gray-100 focus:bg-gray-100 transition duration-100 ease-in-out',
+        error ? 'border border-red-500' : 'border border-Brand',
+      ]"
+      @input="
+        $emit('update:modelValue', ($event.target as HTMLInputElement).value)
+      "
```

Comment:
You don't need to update the value using an input emit. Of course, that will work, but you can achieve the same result in a simpler way. Here's an example from the Vue documentation:
```
<script setup>
const model = defineModel()
</script>

<template>
  <input v-model="model" />
</template>
```